gammapy/utils/random.py

```diff
@@ -78,16 +70,16 @@ def sample_sphere(size, lon_range=None, lat_range=None, unit='rad'):
 
     # Sample random latitude
     v = np.random.random(size)
-    z_range = np.sin(np.array(lat_range))
+    z_range = np.sin(lat_range)
     z = z_range[0] + (z_range[1] - z_range[0]) * v
     # This is not the formula given in the reference, but it is equivalent.
     lat = np.arcsin(z)
 
     # Return result
-    if unit == 'rad':
-        return lon, lat
-    elif unit == 'deg':
-        return np.degrees(lon), np.degrees(lat)
+    if lon_format == '-180_180_deg':
+        # convert back to (-180, 180) deg
+        lon -= Angle(180., 'degree')
```

Comment:
You're re-implementing longitude angle wrapping here, although it's available in Astropy:
http://astropy.readthedocs.org/en/latest/coordinates/angles.html#wrapping-and-bounds

I know you said you tried using `Longitude` objects and there was some issue.
But I'd prefer you work out that issue even if it takes an extra hour or two and we start using Longitude objects in Gammapy.
There's many places already where we need this (catalogs, observation lists) and using `Longitude` objects should make for easier / more readable code.

If you can't get this to work Ok with `Longitude`, can you post a short explanation why?
Could it be solved with a feature request for `Angle` or `Longitude` in Astropy?

In the end, if you conclude that it's just in this specific function that `Longitude` doesn't work for some reason,
OK, I'll still merge as-is.